packages/db/src/lib/execution.ts

```diff
@@ -12,6 +12,34 @@ export interface arnRecord {
 
 const log = new Logger({ sender: '@cumulus/db/lib/execution' });
 
+/**
+ * Returns a list of executionArns sorted by most recent first, for an input
+ * Granule Cumulus ID.
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction
+ *   Knex client for reading from RDS database
+ * @param {number} granuleCumulusId - The primary ID for a Granule
+ * @returns {Promise<arnRecord[]>} - Array of arn objects with the most recent first.
+ */
+export const getExecutionArnsByGranuleCumulusId = (
```

Comment:
Interesting.  For some reason I had thought we were going to drop this from the API returned granule object and/or the work the OXers were doing was covering some of this on the main branch.    *goes looking*

Reply:
I guess the crux of the question is this  - given the current api schema is 

```
    execution: {
      title: 'Step Function Execution link',
      type: 'string',
      readonly: true,
    },
    ```
And the api granule type shows: 

```
export type ApiGranule = {
  granuleId: string
  collectionId: string
  status: GranuleStatus
  execution: string
  cmrLink?: string
  published?: boolean
  pdrName?: string
  provider?: string
  error?: Object
  createdAt: string
  timestamp?: string
  updatedAt: string
  duration?: number
  productVolume?: number
  timeToPreprocess?: number
  timeToArchive?: number
  files?: import('./files').ApiFile[]
} & GranuleTemporalInfo & import('./executions').ExecutionProcessingTimes;
```

Is this actually returning an 'api granule' and if we *are* making changes, is this one we should be returning with the granule?   Apologies if this was already debated, feel free to redirect me to the conversation.

Reply:
Oh was there conversation about it? I didn't see an ADR explicitly calling out anything about the granule response other than "make it look like it used to" but I could be missing something.

I'm including execution ARNs as a list in the response because previously we were returning `execution` as a single ARN string. This seemed like the closest behavior that would support many executions.

It would be cool if we didn't have to bundle executions in the granule response though. A separate query to get executions by `granuleId` would be neat.

Reply:
Also, if we decide to go the way I have it here with a new `executions` array that contains a list of executions ARNs, we'd have to update the `ApiGranule` type, you're totally right. This relates to your other comment here:

https://github.com/nasa/cumulus/pull/2415/files/2e98e42abfc80bdd27ec94433c9e5b824d3a3e7b#r692323145

It's currently not an issue because I'm (wrongly) not using that type as the return type for the translation.

Reply:
I believe this is resolved and the solution was to return the single most recent `execution` as opposed to a list.